src/components/Layout/HeaderCartButton.js

```diff
@@ -67,12 +65,31 @@ const StyledCartButton = styled.button`
 const HeaderCartButton = props => {
    const cartContext = useContext(CartContext);
 
+   const [buttonHighLighted, setButtonHighlighted] = useState(false);
+
    const numCartItems = cartContext.items.reduce((currentNum, item) => {
       return currentNum + item.amount;
    }, 0);
 
+   const {items} = cartContext;
+
+   useEffect(() => {
+      if (items.length === 0) {
+         return;
+      }
+      setButtonHighlighted(true);
```

Comment:
could consider keeping the time somewhere within the component maybe state.
So whenever `setButtonHighlighted` is called, it clears the current timer if any then sets a new one.